fe2-android/app/src/main/java/com/github/dedis/popstellar/ui/settings/SettingsActivity.java

```diff
@@ -0,0 +1,41 @@
+package com.github.dedis.popstellar.ui.settings;
+
+import android.os.Bundle;
+
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.FragmentActivity;
+import androidx.lifecycle.ViewModelProvider;
+
+import com.github.dedis.popstellar.R;
+import com.github.dedis.popstellar.ViewModelFactory;
+import com.github.dedis.popstellar.utility.ActivityUtils;
+
+public class SettingsActivity extends AppCompatActivity {
+
+  private SettingsViewModel mViewModel;
+
+  @Override
+  public void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.settings_activity);
+
+    setupSettingsFragment();
+
+    mViewModel = obtainViewModel(this);
+  }
+
+  public static SettingsViewModel obtainViewModel(FragmentActivity activity) {
+    ViewModelFactory factory = ViewModelFactory.getInstance(activity.getApplication());
+    return new ViewModelProvider(activity, factory).get(SettingsViewModel.class);
+  }
+
+  private void setupSettingsFragment() {
+    SettingsFragment settingsFragment =
+        (SettingsFragment) getSupportFragmentManager().findFragmentById(R.id.fragment_settings);
+    if (settingsFragment == null) {
+      settingsFragment = SettingsFragment.newInstance();
+      ActivityUtils.replaceFragmentInActivity(
+          getSupportFragmentManager(), settingsFragment, R.id.fragment_container_settings);
+    }
```

Comment:
This is a bug I found lately in the rest of the codebase. But this way, if the fragment is found, it will not be displayed.

A PR is coming addressing this issue for the rest of the code.

```suggestion
    }
    ActivityUtils.replaceFragmentInActivity(
        getSupportFragmentManager(), settingsFragment, R.id.fragment_container_settings);
```

Reply:
I don't know if it's necessary, I just thought that fragment were used for the UI and activity existed as a base for them but weren't necessarily UI related.